src/cli/commands/install.js

```diff
@@ -359,6 +359,13 @@ export class Install {
       return true;
     }
 
+    if (!match.integrityMatches && match.integrityError == 'PATTERNS_DONT_MATCH') {
+      this.reporter.warn(this.reporter.lang('integrityPatternsDontMatch'));
+      this.reporter.info(this.reporter.lang('uninstallRegenerate'));
+      this.flags.force = true;
```

Comment:
Probably the logic for missing dependencies should be done not in bailout function but where the lockfile is saved